### PR TITLE
Allow multi col support

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
This PR has the following commits which potentially will make it easy to go through:

1. All the change to the header files
2. Changes to the CDXL operators and Parse Handlers
3. Changes to the CScalarQuantified,ALL, ANY operators
4. Addition of few helpers methods
5. Changes to the translator expr to dxl to translate the sublink testexpr to pass the information of multiple columns used in the subquery clause
6. Most of the changes to the handling of the colrefset
7. Changes to the test files
8. Derivation of PcrsUsed for CScalarQuantified
9. Changes to the minidump

### CODE REVIEW

File: src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp [hunk 1]
```diff
@@ -1596,92 +1615,11 @@ CTranslatorScalarToDXL::TranslateWindowFuncToDXL(
...
-//	@function:
-//		CTranslatorScalarToDXL::CreateScalarCondFromQual
-//
-//	@doc:
-//		Create a DXL scalar boolean operator node from a GPDB qual list.
-//		The function allocates memory in the translator memory pool, and the caller
-//		is responsible for freeing it
-//---------------------------------------------------------------------------
-CDXLNode *
-CTranslatorScalarToDXL::CreateScalarCondFromQual(
```
Comment (bhuvnesh2703) [inline src/backend/gpopt/translate/CTranslatorScalarToDXL.cpp:1614]: These were unused methods

**Review by hardikar — COMMENTED** [review]
Some general comments:
* Splitting out commits by files or file types (header vs source) in a different commit makes it difficult to review, at least for me because it's then hard to follow why specific changes were needed. It is preferable to keep each commit as an independent logical commit (even if it doesn't compile) that captures relevant changes.
* There is now a tool that will auto-format your commit: `/src/tools/fmt fmt`. You'll need to install clang-format & parallel, I believe.
* Can you add an example (output from optimizer_print_plan or optimizer_print_query) of the new query structure?

More comments/questions in the PR.

**Comment by hardikar** [comment]
@bhuvnesh2703 Based on our talk offline, I squashed the logic into a single commit, and updated the PR. I pushed the old commit history at https://github.com/hardikar/gpdb/tree/allow-multi-col-support-before-squash, in case we need it.

**Comment by bhuvnesh2703** [comment]
Closing for now. Will get back to it later.
